### ISSUE TITLE
Problem while computing sd_dipl

### ISSUE THREAD

**Issue by yonizephyr** [honzee/RNAseqCNV#16]
Hi,

I am running this package with the custom snv_format and genome version 38. When I run it on a sample, I get an error of Problem while computing `sd_dipl = sd_dipl`. `sd_dipl` must be size 41 or 1, not 0. I'm not quite sure what this means. When I run it with genome_version 19 I get a slightly different error - Problem while computing `mean_dipl = mean_dipl`.`mean_dipl` must be size 41 or 1, not 0.

On a different sample I instead received this error - Error in `mutate()`: ! Problem while computing `snvOrd = 1:n()`.`snvOrd` must be size 0 or 1, not 2.

I can attach some examples of the files here that are making these errors. Thank you!

[cnv_github.zip](https://github.com/honzee/RNAseqCNV/files/9562121/cnv_github.zip)

**Comment by honzee** [comment]
Hi,

I am sorry for the late reply. I am currently on a vacation and without access to a computer, but I will try to give you as much support as I am able to provide.

First of all thank you for trying out our software, I hope we can get it fixed for you soon.

To help us get to the bottom of it -
1. Please send me the command you used for running the RnaseqCNV
2. Also send me the metadata and condig file
3. Try to change the input from .csv to .tsv format. 

Best,
Jan

**Comment by yonizephyr** [comment]
Thanks for the response! I changed everything to tsv, but reproducing the same error. I'm attaching here the metadata and config files, along with the tsv example files that are making the error. 

The command I used was
`RNAseqCNV_wrapper(config = "path/to/config.txt", metadata = "path/to/metadata.tsv", snv_format = "custom", arm_lvl = FALSE)`
[cnv_github_tsv.zip](https://github.com/honzee/RNAseqCNV/files/9602521/cnv_github_tsv.zip)

**Comment by honzee** [comment]
Great, thank you for replying so quickly. There is an issue with the automated prediction of chromosome copy number.

I suspect it might have something to do with the SNV file. Do you perhaps have a sample that worked or you had only these two, which did not work? If you have a working sample you can try to compare number of SNVs for each chromosome in the files. There may be not enough SNVs after applying all of the filters in the samples, that failed. 

Best,
Jan

**Comment by yonizephyr** [comment]
I do have one sample that worked. Reattaching the zip with those files here.
Interestingly, if you switch out one or the other (either SNV or read_counts) of the one that worked with one of the ones that didn't, it still works, and it doesn't matter which one you switch out. e.g. worked_snv.tsv + sd_dipl_error_read_count.tsv works and sd_dipl_error_snv.tsv + worked_read_counts.tsv together work.

[cnv_github.zip](https://github.com/honzee/RNAseqCNV/files/9619852/cnv_github.zip)

**Comment by honzee** [comment]
I agree, that is interesting. Just a few follow up questions:
1. How does the result from the one working sample look?
2. Is the data from pediatric acute lymphoblastic leukemia or other malignancy? 
3. What is your expected copy number profile of the samples? Do you have other methods for verification? 

P. S. Normally I would be able to check your samples directly, but I will not have access to computer until October. 

Best, 
Jan

**Comment by honzee** [comment]
Dear yonizephyr,

I have finally gone through your files. As I expected, the SNV files are not suitable for RNAseqCNV analysis. It seems that the SNVs were filtered somehow, which is an issue. RNAseqCNV does the necessary filtering internally and ensures that there is enough data left for the analysis. We are not looking for mutations, but for allele variants, that help us determine the final CNV.

As an example, your files have 440 and 473 SNVs. The files that RNAseqCNV was built with have around 500 000 SNVs - while not as many SNVs are kept/needed for the analysis, 500 is probably not enough.

In summary, try to acquire SNV data, that have not gone through filtering steps. If you have any other questions, please let me know; otherwise, I will be closing this issue.

Regards,
Jan

**Comment by yonizephyr** [comment]
Thanks so much for taking a look and apologies for the late reply (now it's my turn to have been away). I actually did the filtering myself, since the dataset I had contained both germline and somatic SNVs. I filtered out the germline, leaving just the somatic SNVs in each tumor sample. Does this R package differentiate between germline and somatic?

**Comment by honzee** [comment]
Not really. The goal is to detect the different patterns of minor allele frequency - so primarily not mutations, but naturally occurring allele variants. The proportion of alleles can guide the CNV detection - for example, if allele frequency is 0.33 - it is highly unlikely, that there are two chromosome copies. For this reason, all germline variants must be kept and not filtered out.

Best,
Jan
